src/components/common/YourCompany/OfficeAddresses.vue

```diff
@@ -567,10 +567,11 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
       this.recMailingAddress = { ...this.getOfficeAddresses?.recordsOffice?.mailingAddress }
       this.recDeliveryAddress = { ...this.getOfficeAddresses?.recordsOffice?.deliveryAddress }
 
+      const hasNoRecAddress = this.isSpecialResolutionFiling
       // set initial validity states
       // these will be updated by the BaseAddress sub-components
-      this.recMailingAddressValid = !isEmpty(this.recMailingAddress)
-      this.recDeliveryAddressValid = !isEmpty(this.recDeliveryAddress)
+      this.recMailingAddressValid = hasNoRecAddress || !isEmpty(this.recMailingAddress)
+      this.recDeliveryAddressValid = hasNoRecAddress || !isEmpty(this.recDeliveryAddress)
```

Comment:
This doesn't look right (compared with other filings where there is no records office).

Maybe `isSpecialResolutionFiling` should be moved from line 547 to line 597?

Reply:
The filing data provided has 
`this.getOfficeAddresses?.registeredOffice` not  `this.getOfficeAddresses?.businessOffice`

Reply:
Leave the API out of it for now. There's different types of addresses and let's leave it at that.

I think that, for coops, they only have Registered Office and no Records Office.

![image](https://user-images.githubusercontent.com/10002889/181818875-cc6b1c45-8949-48b1-b407-cc260763a54a.png)

Maybe you simply need a third case in this method to handle coops. I would prefer that over messing with existing cases. Maybe a helper function could be used to reduce code duplication.

Reply:
Looks good, thanks.